openerp/osv/expression.py

```diff
@@ -913,12 +943,12 @@ def push_result(leaf):
             # -------------------------------------------------
 
             # Applying recursivity on field(one2many)
-            elif column._type == 'one2many' and operator == 'child_of':
+            elif column._type == 'one2many' and operator in parent_child_method.keys():
```

Comment:
It would be nice to extract the list of operators in a constant, e.g.:
```python
HIERARCHICY_OPERATORS = set(('child_of', 'parent_of'))
```
or
```python
HIERARCHY_FUNCS = {'child_of': child_of_domain, 'parent_of': parent_of_domain}
HIERARCHICY_OPERATORS = set(HIERARCHY_FUNCS.keys())
```